### PR TITLE
Feature/admob 2021 Set|Get RequestConfiguration

### PR DESCRIPTION
**Pull request by DellaBitta**
Added functions which allow customers to configure the types of Ads that Admob returns. Whereas this configuration used to be done per AdRequest, this new implementation moves the configuration into the global scope.

  - Implemented the Sets and Gets on Android and iOS.  
  - Created stubs for desktop.
  - Added tests to the integration test framework.
  - Added iOS utility functions to convert NSArray of NSString objects to a vector of std::string objects.

### CODE REVIEW

File: app/src/util_ios.h [hunk 1]
```diff
@@ -192,6 +193,11 @@ void ForEachAppDelegateClass(void (^block)(Class));
 NSMutableArray *StringVectorToNSMutableArray(
     const std::vector<std::string> &vector);
 
+// Convert a NSArray into a vector of strings.  Asserts if a non NSString
+// object is found in the array.
+void NSArrayToStdStringVector(NSArray* array,
+    std::vector<std::string>* string_vector);
```
[the remark this answers is not in the document]
Comment (DellaBitta) [reply app/src/util_ios.h:199]: This file is confusing the linter.  #include <vector> is indeed at the top of this file.

[the remark this answers is not in the document]
Comment (DellaBitta) [reply app/src/util_ios.h:199]: dupe of above.

**Comment by DellaBitta** [comment]
iTest execution: https://github.com/firebase/firebase-cpp-sdk/actions/runs/1238632360

File: app/src/util_ios.h (hunk 1 above)
Comment (jonsimantov) [reply app/src/util_ios.h:199]: Weird, wonder what's up with that.

File: admob/src/include/firebase/admob.h [hunk 2]
```diff
@@ -79,6 +81,24 @@ InitResult Initialize(JNIEnv* jni_env, jobject activity);
...
+void SetRequestConfiguration(const RequestConfiguration& request_configuration);
+
+/// Gets the global RequestConfiguration.
+///
+/// @return the currently active @ref RequestConfiguration that's being
+/// used for every ad request.
+/// @note: on iOS, the
+/// @ref RequestConfiguration::tag_for_child_directed_treatment and
+/// @ref RequestConfiguration::tag_for_under_age_of_consent fields will be set
+/// set to kChildDirectedTreatmentUnspecified, and
```
Comment (jonsimantov) [inline admob/src/include/firebase/admob.h:98]: nit: "set" is written twice.

File: app/src/util_ios.h [hunk 3]
```diff
@@ -192,6 +193,11 @@ void ForEachAppDelegateClass(void (^block)(Class));
 NSMutableArray *StringVectorToNSMutableArray(
     const std::vector<std::string> &vector);
 
+// Convert a NSArray into a vector of strings.  Asserts if a non NSString
+// object is found in the array.
+void NSArrayToStdStringVector(NSArray* array,
```
Comment (jonsimantov) [inline app/src/util_ios.h:198]: soft suggested name: NSArrayOfNSStringToVectorOfString

File: admob/src/ios/FADRequest.mm [hunk 4]
```diff
@@ -24,16 +24,6 @@
 GADRequest *GADRequestFromCppAdRequest(AdRequest adRequest) {
   // Create the GADRequest.
   GADRequest *request = [GADRequest request];
-
-  // Test devices.
-  if (adRequest.test_device_id_count > 0) {
```
Comment (jonsimantov) [inline admob/src/ios/FADRequest.mm:29]: Are these set elsewhere now?

Comment (DellaBitta) [reply admob/src/ios/FADRequest.mm:29]: Yes, they were once per ad request, now they're part of RequestConfiguration structure. set globally through SetRequestConfigration().  No need to set them each and every time anymore!

File: app/src/util_ios.h (hunk 3 above)
Comment (DellaBitta) [reply app/src/util_ios.h:198]: Will do.

File: admob/src/include/firebase/admob.h (hunk 2 above)
Comment (DellaBitta) [reply admob/src/include/firebase/admob.h:98]: Thanks, will fix.

Comment (DellaBitta) [reply admob/src/include/firebase/admob.h]: Done.

File: app/src/util_ios.h (hunk 3 above)
Comment (DellaBitta) [reply app/src/util_ios.h]: Done.

File: app/src/util_ios.h [hunk 5]
```diff
@@ -192,6 +193,11 @@ void ForEachAppDelegateClass(void (^block)(Class));
 NSMutableArray *StringVectorToNSMutableArray(
     const std::vector<std::string> &vector);
 
+// Convert a NSArray into a vector of strings.  Asserts if a non NSString
+// object is found in the array.
+void NSArrayOfNSStringToVectorOfString(NSArray* array,
+    std::vector<std::string>* string_vector);
```
[the remark this answers is not in the document]
Comment (DellaBitta) [reply app/src/util_ios.h:199]: This file is confusing the linter.  #include <vector> is indeed at the top of this file.

**Review by jonsimantov — APPROVED** [review]

**Pull request merged**